static/app/components/searchSyntax/parser.tsx

```diff
@@ -600,9 +598,103 @@ export class TokenConverter {
       return this.checkInvalidInFilter(value as InFilter['value']);
     }
 
+    if (
+      [
+        FilterType.AggregateDate,
+        FilterType.AggregateDuration,
+        FilterType.AggregateNumeric,
+        FilterType.AggregatePercentage,
+        FilterType.AggregateRelativeDate,
+      ].includes(filter)
+    ) {
+      return this.checkInvalidAggregateFilter(
+        filter as AggregateFilter,
+        key as FilterMap[AggregateFilter]['key'],
+        value as FilterMap[AggregateFilter]['value']
+      );
+    }
+
     return null;
   };
 
+  checkInvalidAggregateFilter = (
+    filterType: AggregateFilter,
+    key: FilterMap[AggregateFilter]['key'],
+    value: FilterMap[AggregateFilter]['value']
+  ) => {
+    const {isNumeric, isDuration, isDate, isPercentage} = this.keyValidation;
+
+    const keyName = key.name.text;
+
+    const checkAggregateKey = (check: (s: string) => boolean) => {
+      if (check(keyName)) {
+        if (key.args === null) {
+          // Aggregate keys without args will rely on default values
+          return null;
+        }
+
+        // Get valid column types from the aggregate key
+        const aggregation = AGGREGATIONS[keyName];
+        const validColumnTypes = (aggregation?.parameters?.find(p => p.kind === 'column')
+          ?.columnTypes ?? null) as ValidateColumnTypes | null;
+
+        // Check each argument whether it is a valid parameter to this aggregation
+        const invalidArgs = key.args.args.filter(arg => {
+          const argKey = arg?.value?.value;
+          if (!argKey) {
+            return true;
+          }
+
+          const fieldDef = getFieldDefinition(argKey);
+          if (!fieldDef?.valueType) {
+            return true;
+          }
+
+          return !validateAggregateType(
+            validColumnTypes,
+            argKey,
+            fieldDef.valueType as ColumnType
+          );
+        });
+
+        if (invalidArgs.length > 0) {
+          return {
+            reason: getAggregateInvalidArgumentMessage(
+              keyName,
+              validColumnTypes,
+              invalidArgs.map(arg => arg.value?.value ?? '')
+            ),
+          };
+        }
+
+        return null;
+      }
+
+      const definition = getFieldDefinition(keyName);
+      return {
+        reason: `'${keyName}' returns a ${
+          `${definition?.valueType}` ?? 'different value type.'
+        }; '${value.text}' is not valid here.`,
+      };
+    };
+
+    switch (filterType) {
+      case FilterType.AggregateRelativeDate:
+      case FilterType.AggregateDate:
+        return checkAggregateKey(isDate.bind(this));
```

Comment:
Why do these need bound?

Reply:
Huh, I don't recall why, seems weird. I'll remove it